src/core/createLogger.js

```diff
@@ -10,38 +10,107 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-export default (console, getDebugEnabled, prefix) => {
-  const process = (level, ...rest) => {
+import { assign } from "../utils";
+
+export default ({ getDebugEnabled, console, getMonitors, context }) => {
+  let prefix = `[${context.instanceName}]`;
+  if (context.componentName) {
+    prefix += ` [${context.componentName}]`;
+  }
+
+  const notifyMonitors = (method, data) => {
+    const monitors = getMonitors();
+    if (monitors.length > 0) {
+      const dataWithContext = assign({}, context, data);
+      monitors.forEach(monitor => {
+        if (monitor[method]) {
+          monitor[method](dataWithContext);
+        }
+      });
+    }
+  };
+
+  const log = (level, ...rest) => {
     if (getDebugEnabled()) {
       console[level](prefix, ...rest);
     }
   };
 
   return {
     get enabled() {
-      return getDebugEnabled();
+      return getMonitors().length > 0 || getDebugEnabled();
+    },
+    logOnInstanceCreated(data) {
+      notifyMonitors("onInstanceCreated", data);
+      log("info", "Instance initialized.");
+    },
+    logOnInstanceConfigured(data) {
+      notifyMonitors("onInstanceConfigured", data);
+      log("info", "Instance configured. Computed configuration:", data.config);
+    },
+    logOnBeforeCommand(data) {
+      notifyMonitors("onBeforeCommand", data);
+      log(
+        "info",
+        `Executing ${data.commandName} command. Options:`,
+        data.options
+      );
```

Comment:
Considering this will log a message to the console, shouldn't monitors' `onBeforeLog` also be called? Maybe I'm overlooking where that's happening.

Reply:
I was thinking that the monitor was already notified of the event so then didn't need the log notification as well.  I think you are right though.